File: c++/basic_grammar/operation/assignment_operator/division.cpp
#include <iostream>
using namespace std;

int division_assignment_operator() {
  int a = 64;
  for (int i = 0; i < 5; i++) {
    a /= 2;
  }
  return a;
}

double decimal_division_assignment_operator() {
  double a = 243;
  for (int i = 0; i < 5; i++) {
    a /= 1.5;
  }
  return a;
}

int negative_division_assignment_operator() {
  int a = 64;
  for (int i = 0; i < 5; i++) {
    a /= -2;
  }
  return a;
}

int main() {
  cout << division_assignment_operator() << endl;
  cout << decimal_division_assignment_operator() << endl;
  cout << negative_division_assignment_operator() << endl;
  return 0;
}

File: typescript/basic_grammar/operation/assignment_operator/division.ts
export const divisionAssignmentOperator = () => {
  let a = 64
  for (let i = 0; i <  5; i++) {
    a /= 2
  }
  return a
}

export const decimalDivisionAssignmentOperator = () => {
  let a = 243
  for (let i = 0; i < 5; i++) {
    a /= 1.5
  }
  return a
}

export const negativeDivisionAssignmentOperator = () => {
  let a = 64
  for (let i = 0; i < 5; i++) {
    a /= -2
  }
  return a
}

console.log(divisionAssignmentOperator())
console.log(decimalDivisionAssignmentOperator())
console.log(negativeDivisionAssignmentOperator())
